cmd/infra/azure/create.go

```diff
@@ -166,41 +170,48 @@ func (o *CreateInfraOptions) Run(ctx context.Context, l logr.Logger) (*CreateInf
 	}
 	l.Info("Successfully assigned contributor role to managed identity", "name", identityID)
 
-	securityGroupID := ""
-
-	if o.NetworkSecurityGroup != "" {
-		//check that the provided network security group exists
-		securityGroupClient, err := armnetwork.NewSecurityGroupsClient(subscriptionID, azureCreds, nil)
+	// Retrieve a client's existing virtual network if a client resource group was provided; otherwise, create a new VNET with a network security group
+	if len(o.ClientResourceGroupName) > 0 {
+		vnet, err := getClientsVirtualNetwork(ctx, subscriptionID, o.ClientResourceGroupName, o.ClientVnetName, azureCreds)
 		if err != nil {
-			return nil, fmt.Errorf("failed to create security group client: %w", err)
+			return nil, err
 		}
-		nsg, err := securityGroupClient.Get(ctx, resourceGroupName, o.NetworkSecurityGroup, nil)
-		if err != nil {
-			return nil, fmt.Errorf("failed to get network security group: %w", err)
+
+		result.SubnetID = *vnet.Properties.Subnets[0].ID
+		result.SubnetName = *vnet.Properties.Subnets[0].Name
```

Comment:
nit: similarly, length checks on array access would keep us from panicking and crashing

Reply:
I added checks.